addons/payment_sips/models/sips.py

```diff
@@ -0,0 +1,259 @@
+# -*- coding: utf-'8' "-*-"
+
+try:
+    import simplejson as json
+except ImportError:
+    import json
+import logging
+from hashlib import sha256
+import urlparse
+import unicodedata
+
+from openerp.tools.float_utils import float_compare
+from openerp import models, fields, api
+from openerp.exceptions import ValidationError
+from openerp.addons.payment.models.payment_acquirer import ValidationError
+from openerp.addons.payment_sips.controllers.main import SipsController
+
+_logger = logging.getLogger(__name__)
+
+
+UNORMALIZE_CHARS = {
+    u'Š': u'S', u'š': u's', u'Đ': u'Dj', u'đ': u'dj', u'Ž': u'Z', u'ž': u'z',
+    u'Č': u'C', u'č': u'c', u'Ć': u'C', u'ć': u'c', u'À': u'A', u'Á': u'A', u'Â': u'A', u'Ã': u'A',
+    u'Ä': u'A', u'Å': u'A', u'Æ': u'A', u'Ç': u'C', u'È': u'E', u'É': u'E', u'Ê': u'E', u'Ë': u'E',
+    u'Ì': u'I', u'Í': u'I', u'Î': u'I', u'Ï': u'I', u'Ñ': u'N', u'Ò': u'O', u'Ó': u'O', u'Ô': u'O',
+    u'Õ': u'O', u'Ö': u'O', u'Ø': u'O', u'Ù': u'U', u'Ú': u'U', u'Û': u'U', u'Ü': u'U', u'Ý': u'Y',
+    u'Þ': u'B', u'ß': u'Ss', u'à': u'a', u'á': u'a', u'â': u'a', u'ã': u'a', u'ä': u'a', u'å': u'a',
+    u'æ': u'a', u'ç': u'c', u'è': u'e', u'é': u'e', u'ê': u'e', u'ë': u'e', u'ì': u'i', u'í': u'i',
+    u'î': u'i', u'ï': u'i', u'ð': u'o', u'ñ': u'n', u'ò': u'o', u'ó': u'o', u'ô': u'o', u'õ': u'o',
+    u'ö': u'o', u'ø': u'o', u'ù': u'u', u'ú': u'u', u'û': u'u', u'ý': u'y', u'ý': u'y', u'þ': u'b',
+    u'ÿ': u'y', u'Ŕ': u'R', u'ŕ': u'r', u"`": u"'", u"´": u"'", u"„": u",", u"`": u"'",
+    u"´": u"'", u"“": u"\"", u"”": u"\"", u"´": u"'", u"&acirc;€™": u"'", u"{": u"",
+    u"~": u"", u"–": u"-", u"’": u"'", u">": u" ", u"<": u" "
+}
+
+CURRENCY_CODES = {
+    'EUR': '978',
+    'USD': '840',
+    'CHF': '756',
+    'GBP': '826',
+    'CAD': '124',
+    'JPY': '392',
+    'MXN': '484',
+    'TRY': '949',
+    'AUD': '036',
+    'NZD': '554',
+    'NOK': '578',
+    'BRL': '986',
+    'ARS': '032',
+    'KHR': '116',
+    'TWD': '901',
+}
+
+
+def unormalize(text):
```

Comment:
I just noticed but this method is actually never used, so we don't really need it anymore.